source/develop/release-management/features/virt/parallel-migration-connections.md

```diff
@@ -0,0 +1,88 @@
+
+## Prerequisites
+
+Nothing special, just up-to-date libvirt and QEMU.
+
+## Limitations
+
+Current implementation of parallel migration connections in QEMU has some restrictions:
+
+- It doesn't support switching to post-copy.
+
+- RDMA doesn't work.  This is not a limitation for oVirt, because RDMA is not supported in oVirt.
+
+- XBZRLE compression is not supported.  However this kind of compression is generally not recommended to use in migrations because it adds little benefit and a significant overhead.  gzip/zstd is supported instead and migrations should actually benefit from it.  (TODO: Check whether libvirt supports this kind of compression.)
+
+- Due to their better implementation, parallel migration connections can be faster than the traditional migration implementation even when the network bandwidth can be saturated with a single connection.  However, when single connection is specified, the parallel migration connection implementation uses a different code path, which may not be sufficiently robust, and it's recommended to always use at least two connections.  In the worst case, one of the threads will be idle while causing only a little overhead. 
+
+- Parallel migration connections haven't been tested extensively with connections faster than 100 Gbps.  Due to current multifd implementation, there is an increasing overhead when using many connections and it's not clear whether using more than 16 parallel connections adds a real benefit.
+
+- It still applies that large guests with terrabytes of RAM are difficult to migrate successfully.
+
+- Since it's easier to saturate network bandwidth with multiple connections, it's advisable to not migrate multiple VMs from a single host at the same time.
+
+- VM CPU pinning may restrict the number of CPUs available to the QEMU process and thus impose a limit on the meaningful number of parallel connections.
+
+- It's difficult to guess the right number of parallel connections to use.  A rule of thumb is to use one connection per 10 Gbps of available network bandwidth.  If the number of connections is a bit high, some threads may remain idle while the related overhead should be negligible.
+
+- There is very little documentation about parallel migration connections and the information above was obtained directly from platform developers.
+
+## User experience
+
+A new migration option to enable parallel migration connections is introduced in cluster and VM edit dialogs.  The cluster option specifies what to use in the given cluster.  VMs can override the cluster value individually.  The new option offers the following choices:
+
+- **Disabled** (the default in clusters): Parallel migration connections are not used.
+
+- **Auto**: Use parallel migration connections (TODO: Even when there is insufficient network bandwidth, for the supposed benefit of a more efficient implementation?) and set their number automatically based on the available resources (network bandwidth and CPU power).
```

Comment:
about the TODO - good question, probably yes but I think it should appear in the document in a more general way (that implies parallel migration may also not be set)

Reply:
OK, I'll list both the options here (always parallel x only if there are sufficient resources).

Reply:
Done.